manifests/init.pp

```diff
@@ -249,4 +254,14 @@
   class { 'candlepin::artemis': } ~>
   class { "candlepin::database::${candlepin::db_type}": } ~>
   Class['candlepin::service']
+
+  # Add the candlepin.consumer.facts.match_regex option if provided
+  if $facts_match_regex {
+    file_line { 'candlepin_facts_match_regex':
```

Comment:
We already manage the file `$candlepin_conf_file` so you must update the template. You can't make it idempotent this way: the file resource will revert the file_line and then file_line will place it back. So every single time it will apply changes, which will result in restarting the service.

We also put all our configuration in the `candlepin::config` class and use the `candlepin` class just to order the resources properly.